server/handler/redirect.go

```diff
@@ -19,6 +19,7 @@ import (
 func RedirectToHTTPS(w http.ResponseWriter, req *http.Request, redirectStatusCode int) {
 	targetURL := req.URL
 	targetURL.Scheme = "https"
+	targetURL.Host = req.Host
```

Comment:
add a comment everywhere about this. golang limitation on req.URL.Host

Reply:
https://github.com/golang/go/issues/28940

Reply:
https://stackoverflow.com/questions/42921567/what-is-the-difference-between-host-and-url-host-for-golang-http-request